### ISSUE TITLE
Recaptcha is skipped

### ISSUE THREAD

**Issue by lilmike** [spbriggs/recaptcha-plugin#5]
On my website,  1u.rs,  The captcha is totally skipped. It shows on the front page, but even if leaving the box unchecked, it still adds the link. Any idea?
Michael.

**Comment by spbriggs** [comment]
Hi. It looks like you're using the new Invisible Recaptcha. This is untested and unsupported at the moment. Can you try getting a Recaptcha v2 key and see if that works?

**Comment by lilmike** [comment]
Hi,
Two things:

1. I fixed the recaptcha skipping by adding the fields status => fail, code => error: captcha, and errorCode => 403. I did this buy, instead of returning directly the $response variable, whether from solvemedia or recaptcha, waiting until the bottom to return it.
The code looks like this:

if (isset($response['message'])) {
$response['status'] = 'fail';
$response['error'] = 'error: captcha';
$response['errorCode'] = '403';
return $response;
}
else {
return false;
}

2. I added the following to support invisible recaptcha:


echo <<<HTML
<script>
function submit_form() {
document.getElementById('URL-Form').submit();
}
</script>

And in the echo that usually would add the div, add the following attributes:

data-bind='recaptcha-submit'
data-callback='submit_form'


In the homepage:


in the input for submit add the attribute:

id='recaptcha-submit'

and the form tag:

id='URL-Form'


That's all I needed to do to get it working, besides getting new invisible recaptcha keys. Obviously it could be more featureful (for example allowing the admin to choose v2 or v3 on the options page, but this is only for my own use so :-))

-Michael.
